src/test/java/hudson/plugins/git/GitChangeSetBasicTest.java

```diff
@@ -130,4 +134,77 @@ public void testSwedishDate() {
     public void testSwedishTimestamp() {
         assertEquals(1363875404000L, genChangeSetForSwedCase(true).getTimestamp());
     }
+
+    @Test
+    public void testChangeLogTruncationWithShortMessage(){
+        GitChangeSet changeSet = GitChangeSetUtil.genChangeSet(true, false, true,
+                "Lorem ipsum dolor sit amet.",
+                true);
+        String msg = changeSet.getMsg();
+        Assert.assertTrue("title is correct", msg.contains("Lorem ipsum dolor sit amet."));
+        Assert.assertTrue("title has correct size", msg.length() <= GitChangeSet.TRUNCATE_LIMIT);
+    }
+    @Test
+    public void testChangeLogTruncationWithNewLine(){
+        GitChangeSet changeSet = GitChangeSetUtil.genChangeSet(true, false, true,
+                "Lorem ipsum dolor sit amet, "+System.lineSeparator()+"consectetur adipiscing elit.",
+                true);
+        String msg = changeSet.getMsg();
+        Assert.assertTrue("title is correct", msg.contains("Lorem ipsum dolor sit amet,"));
+        Assert.assertTrue("title has correct size", msg.length() <= GitChangeSet.TRUNCATE_LIMIT);
+    }
+
+    @Test
+    public void testChangeLogTruncationWithoutNewLine(){
+        GitChangeSet changeSet = GitChangeSetUtil.genChangeSet(true, false, true,
+                "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus pellentesque ipsum non aliquam interdum. Integer metus orci, vulputate id turpis in, pharetra pretium magna. Fusce sollicitudin vehicula lectus. Nam ut eros purus. Mauris aliquam mi et nunc porta, non consectetur mauris pretium. Fusce a venenatis dolor. Sed commodo, dui ac posuere dignissim, dolor tortor semper eros, varius consequat nulla purus a lacus. Vestibulum egestas, orci vitae pellentesque laoreet, dolor lorem molestie tellus, nec luctus lorem ex quis orci. Phasellus interdum elementum luctus. Nam commodo, turpis in sollicitudin auctor, ipsum lectus finibus erat, in iaculis sapien neque ultrices sapien. In congue diam semper tortor laoreet aliquet. Mauris lacinia quis nunc vel accumsan. Nullam sed nisl eget orci porttitor venenatis. Lorem ipsum dolor sit amet, consectetur adipiscing elit",
+                true);
+        String msg = changeSet.getMsg();
+        Assert.assertTrue("title is correct", msg.contains("Lorem ipsum dolor sit amet, consectetur adipiscing elit."));
```

Comment:
Hamcrest here as well.

Also, please add an assertion that the message does not contain the full string.  Since the assertion is testing a subset of the string, that assertion would pass even if the full message were returned.